articles/introduction-to-constructors-in-c++/index.md

```diff
@@ -0,0 +1,270 @@
+- Constructors should be declared in the public section of the Class.
+- Constructors cannot return values to the calling program because they do not have return types.
+- Constructors should always be non-virtual. They are static functions and in C++ so they cannot be virtual.     
+- Constructors cannot be inherited because they are called when objects of the class are created. It is therefore not realistic to create a base class object using the derived class constructor function.
+- We cannot refer to the addresses of constructors since we only require the address of a function to call it and they can never be called directly. Also, the language does not permit it.
+
+**NOTE: _Whenever a Constructor is declared, initialization of the class objects becomes imperative._**
+
+#### Types of constructors
+**1. Default constructors**
+
+They are constructors that do not pass or accept any parameters. The default constructor for a class C is `C::C()`. The compiler supplies a default constructor for instances where it is not defined.
+
+Therefore a statement such as `integer v1` invokes the default constructor of the compiler to create the object v1.
+
+ **Example of default constructor**
+```c++
+// c++ program to illustrate the concept of constructors
+
+#include <iostream>
+using namespace std;
+
+class integer
+{
+public:
+  int x, y;
+
+// Default Constructor declared
+  integer()
+  {
+    x = 45;
+    y = 10;
+  }
+};
+
+int main()
+{
+  // Default constructor called automatically when the object is created
+  integer a;
+  cout << "x: " << a.x << endl << "y: " << a.y;
+  return 0;
+}
+```
+In the program above, the compiler will automatically provide a default constructor implicitly. Our program output will be:
+
+```bash
+x: 45
+y: 10
+```
+
+**2. Parameterized constructors**
+
+In practice, we may be required to initialize the data elements of different objects with different values. C++ enables us to accomplish this by the use of parameterized constructors which can take parameters when the objects are created
+
+In the program below we are going to illustrate how our constructor `integer()` will be.
+
+```c++
+// C++ program to illustrate parameterized constructors
+
+#include <iostream>
+using namespace std;
+
+class integer
+{
+private:
+  int a, b;
+public:
+  // Parameterized Constructor
+  integer(int x, int y)
+  {
+    a = x;
+    b = y;
+  };
+
+  int getX()
+  {
+    return a;
+  };
+
+  int getY()
+  {
+    return b;
+  };
+};
+int main()
+{
+  integer v(10, 15); // Constructor called
+
+  cout << "a = " << v.getX() << endl; // values assigned by constructor accessed
+  cout<< "b = " << v.getY() << endl;
+
+  return 0;
+}
+```
+
+Initial values have to be passed to the object through parameterized constructors and the normal declaration of a statement will not work. We can pass the initial values as arguments by either calling the constructor implicitly or explicitly.
```

Comment:
I'd suggest you give the expected output of the code above this since it is an entire cpp program.

The output is :
```bash
a = 10
b = 15
```